pkg/services/store/object/sqlstash/sql_storage_server.go

```diff
@@ -0,0 +1,220 @@
+package sqlstash
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/grpcserver"
+	"github.com/grafana/grafana/pkg/services/sqlstore/db"
+	"github.com/grafana/grafana/pkg/services/sqlstore/session"
+	"github.com/grafana/grafana/pkg/services/store/object"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+func ProvideSQLObjectServer(db db.DB, cfg *setting.Cfg, grpcServerProvider grpcserver.Provider) object.ObjectStoreServer {
+	objectServer := &sqlObjectServer{
+		sess: db.GetSqlxSession(),
+		log:  log.New("in-memory-object-server"),
+	}
+	object.RegisterObjectStoreServer(grpcServerProvider.GetServer(), objectServer)
+	return objectServer
+}
+
+type sqlObjectServer struct {
+	log  log.Logger
+	sess *session.SessionDB
+}
+
+func (s sqlObjectServer) Read(ctx context.Context, r *object.ReadObjectRequest) (*object.ReadObjectResponse, error) {
+	return nil, fmt.Errorf("not implemented yet")
+}
+
+func (s sqlObjectServer) BatchRead(ctx context.Context, batchR *object.BatchReadObjectRequest) (*object.BatchReadObjectResponse, error) {
+	return nil, fmt.Errorf("not implemented yet")
+}
+
+func createContentsHash(contents []byte) string {
+	hash := md5.Sum(contents)
+	return hex.EncodeToString(hash[:])
+}
+
+func userFromContext(ctx context.Context) *user.SignedInUser {
+	// TODO implement in GRPC server
+	return &user.SignedInUser{
+		UserID: 1,
+		OrgID:  1,
+		Login:  "fake",
+	}
+}
+
+func (s sqlObjectServer) Write(ctx context.Context, r *object.WriteObjectRequest) (*object.WriteObjectResponse, error) {
+	// TODO: this needs to be extracted from the body content
+	summary := object.ObjectSummary{
+		Name:        "hello",
+		Description: "description",
+		Labels: map[string]string{
+			"hello": "world",
+			"test":  "",
+		},
+		References: []*object.ExternalReference{
+			{
+				Kind: "ds",
+				Type: "influx",
+				UID:  "xyz",
+			},
+		},
+	}
+
+	etag := createContentsHash(r.Body)
+	summaryjson, err := json.Marshal(summary)
+	if err != nil {
+		return nil, err
+	}
+
+	modifier := userFromContext(ctx)
+	key := fmt.Sprintf("%d/%s.%s", modifier.OrgID, r.UID, r.Kind)
+
+	rsp := &object.WriteObjectResponse{}
+
+	err = s.sess.WithTransaction(ctx, func(tx *session.SessionTx) error {
```

Comment:
@zserge -- any red flags with this approach?

I am still trying to learn how to best use sqlx here :)